### PR TITLE
Users index endpoint

### PR DESCRIPTION
**Pull request by mnmallea**
## Summary
- Added users index endpoint.

## Known Issues
None.

## Trello Card
https://trello.com/c/uh3Esni5/3-lista-de-usuarios

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 const { healthCheck } = require('./controllers/healthCheck');
-const { createUser } = require('./controllers/users');
+const { createUser, usersIndex } = require('./controllers/users');
 const { validateUserEmailUniqueness } = require('./middlewares/users');
 const { userCreationSchema } = require('./schemas/users');
 const { schemaValidation } = require('./middlewares/fields_validation');
 
 exports.init = app => {
   app.get('/health', healthCheck);
+  app.get('/users', usersIndex);
```
Comment (lnmedrano) [inline app/routes.js:9]: User should be authenticated to consume this endpoint.

File: app/services/users.js [hunk 2]
```diff
@@ -3,3 +3,6 @@ const { User } = require('../models');
 exports.createUser = userParams => User.create(userParams);
 
 exports.findUserByEmail = email => User.findOne({ where: { email } });
+
+exports.findAndCountAllUsers = ({ offset, limit }) =>
+  User.findAndCountAll({ offset, limit, order: [['id', 'asc']] });
```
Comment (lnmedrano) [inline app/services/users.js:8]: Ordering is not necessary. The card does not asked for it.

File: app/controllers/users.js [hunk 3]
```diff
@@ -10,3 +11,8 @@ exports.createUser = (req, res, next) => {
     .then(user => res.status(201).send(showUserSerializer(user)))
     .catch(next);
 };
+
+exports.usersIndex = (req, res, next) =>
```
Comment (lnmedrano) [inline app/controllers/users.js:15]: I think getUsers is a more appropriate name, but it's ok :+1:

File: test/users/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      usersIndexResponse = await httpRequest();
+    });
+
+    test('Responds with 200 status code', () => expect(usersIndexResponse.statusCode).toBe(200));
+
+    test('Responds with the expected body structure', () =>
+      expect(usersIndexResponse.body).toMatchObject({
+        page: expect.any(Array),
+        total_count: expect.any(Number)
+      }));
```
Comment (lnmedrano) [inline test/users/index.spec.js:29]: Remember that the `toHaveExpectedProperty` could be used here.

File: test/users/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    test('Responds with the expected users', () => {
+      expect(usersIndexResponse.body.page).toEqual([showUserSerializer(users[1])]);
+    });
+  });
+});
```
Comment (lnmedrano) [inline test/users/index.spec.js:52]: There are paths that are not being tested:
- What if page is a string?
- What if limit is a string?
- What if both are strings?
- What if the user is not authenticated?
- What if I ask for a page with no users?

File: app/services/users.js (hunk 2 above)
Comment (mnmallea) [reply app/services/users.js:8]: But without a default order, rows could be retrieved on different order when iterating throught pages

File: app/controllers/users.js (hunk 3 above)
Comment (mnmallea) [reply app/controllers/users.js:15]: :rails:

File: test/users/index.spec.js (hunk 5 above)
Comment (mnmallea) [reply test/users/index.spec.js:88]: 1, 2, 3 --> field validation error
4 --> 401
5 --> empty page

File: test/users/index.spec.js (hunk 4 above)
Comment (mnmallea) [reply test/users/index.spec.js]: I took a look of this, but with `toMatchObject` I can test the return types.

Comment (lnmedrano) [reply test/users/index.spec.js]: Yes. But we can make that two validations in separate tests. If this test fails I don't know if the field is not present in the response or if it has a different type

**Pull request merged**